### PR TITLE
make it possible to use a git remote other than "origin"

### LINKED ISSUE #88: Ability to specify which remote name to use in `use_travis_deploy()`
Currently `use_travis_deploy()` and in particular `github_info()` seems to be hardcoded to use the 'origin' remote. It would be useful to be able to specify a specific remote name to use rather than unconditionally using origin if it exists. e.g. I had an 'origin' remote to my fork of a repo, but I actually wanted to create deploy keys to the 'upstream' remote.

I tried specifying the repo explicitly with the `repo` and `travis_repo` arguments, but the `info` argument is still used  by `use_travis_deploy()` in the call to `github_add_key()`.

I ended up working around the issue by temporarily renaming my remotes to do what I wanted.

### PR DESCRIPTION
**Pull request by pat-s**
fixes #88 

Added one test for `github_info()` where `remote` is actually used.

### CODE REVIEW

File: tests/testthat/test-custom-remote.R [hunk 1]
```diff
@@ -0,0 +1,12 @@
+context("remote")
+
+setwd("./travis-testthat")
```
Comment (krlmlr) [inline tests/testthat/test-custom-remote.R]: This looks dangerous. Can we use `withr::local_dir()` inside the test instead?

File: tests/testthat/test-custom-remote.R [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+test_that("custom remote is honored", {
+
+  system("git remote add test https://github.com/pat-s/travis-testthat.git")
+
+  expect_is(github_info(remote = "test"), "gh_response")
```
Comment (krlmlr) [inline tests/testthat/test-custom-remote.R]: `expect_s3_class()` is now the recommended test.

File: R/github-key.R [hunk 3]
```diff
@@ -80,11 +86,14 @@ create_key_data <- function(pubkey, title) {
 #' @param info `[list]`\cr
 #'   GitHub information for the repository, by default obtained through
 #'   [github_info()].
+#' @template remote
 #'
 #' @export
 #' @keywords internal
 #' @rdname github_info
-github_repo <- function(path = usethis::proj_get(), info = github_info(path)) {
+github_repo <- function(path = usethis::proj_get(),
```
Comment (krlmlr) [inline R/github-key.R]: Can we adopt the usethis model of the "current project" here (after merging this PR)?

File: tests/testthat/test-custom-remote.R (hunk 1 above)
Comment (pat-s) [reply tests/testthat/test-custom-remote.R]: Replaced all `setwd()´ by 

```r
withr::with_dir(
    here::here("tests/testthat/travis-testthat"),
    {

```

File: tests/testthat/test-custom-remote.R (hunk 2 above)
Comment (pat-s) [reply tests/testthat/test-custom-remote.R]: Done.

File: R/github-key.R (hunk 3 above)
Comment (pat-s) [reply R/github-key.R]: `package = NULL`?

File: tests/testthat/test-custom-remote.R (hunk 1 above)
Comment (krlmlr) [reply tests/testthat/test-custom-remote.R]: I still like `local_dir()` much better, because this reduces indent.

**Comment by krlmlr** [comment]
Thanks!

**Pull request merged**